### DISCUSSION TITLE
How to map the id of the gptcache question table and the service request id

### DISCUSSION THREAD

**Discussion by qhdrl12** [zilliztech/GPTCache#615] [General]
I have a need to view gptcache's question table by joining the service's history data.

It would be convenient if gptcache_question.id and my service.id are the same, but currently gptcache_question.id does not appear to work due to external settings.

Please advise on this task.

**Comment by SimFG** [comment]
what's the meaning of `service.id`

**Reply by qhdrl12** [reply]
This is the simple transaction ID I use.

**Reply by SimFG** [reply]
you expect that the question_id can be customized outside, right?

**Reply by qhdrl12** [reply]
I'm asking because I don't think question_id can be customized. 

But is there any way to set it up?

I think there might be a need to see gptcache_question.id and the request history of the llm system together.

**Reply by SimFG** [reply]
@qhdrl12 if you want to see the question id and content, maybe you can connect the scalar db (sqlite, mysql, etc) and query them

**Reply by qhdrl12** [reply]
that's right. I checked the question content in mysql.
![image](https://github.com/zilliztech/GPTCache/assets/15104468/9b350488-d75f-4266-bcaa-d432898792cb)

Well, and separately from gptcache question, I am using the question table used by the service, and the table is stored in the following format.
![image](https://github.com/zilliztech/GPTCache/assets/15104468/049a086f-95cf-47d9-8b3e-da17585d2cba)

i would like to join these two tables, but the question and query contain the same data, but handling duplicate questions is not possible.

**Reply by SimFG** [reply]
When adding cache data, providing a callback may meet your needs?

**Reply by qhdrl12** [reply]
yes! I think this requirement would be met if a callback was provided that provided the Question.id (or some additional data) when adding the cache.

**Reply by SimFG** [reply]
got it🤝

**Reply by qhdrl12** [reply]
Thank you for your valuable time! 👍
